File: storage/src/main/java/com/seamstressless/storage/domain/entities/Supply.java
package com.seamstressless.storage.domain.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Data
@EqualsAndHashCode(callSuper = true)
@Table(name = "supplies")
public class Supply extends BaseEntity<Long> {

    @Column(name = "url", nullable = false)
    private String URL;

    @Column(name = "store_name", nullable = false)
    private String storeName;

    @Column(name = "x_path_price", nullable = false)
    private String xPathPrice;

    @ManyToOne
    @JoinColumn(name = "category_id", nullable = false)
    private SupplyCategory category;
}

File: SeamStressLess/src/app/feature/supply/models/Supply.ts
import {SupplyCategory} from "./SupplyCategory";

export interface Supply {
  id: number;
  storeName: string;
  URL: string;
  XPathPrice: string;
  Category: SupplyCategory;
  lastPrice: number;
}

export interface SupplyPost {
  URL: string;
  storeName: string;
  XPathPrice: string;
  categoryId: number;
}
